src/libs/ck-libs/io/ckio.h

```diff
@@ -70,11 +82,49 @@ namespace Ck { namespace IO {
   /// Close a previously-opened file. All sessions on that file must have
   /// already signalled that they are complete.
   void close(File file, CkCallback closed);
+  
+  /**
+   * Prepare to read data from @arg file section specified by @arg bytes and @arg offset.
+   * This method will proceed to eagerly read all of the data in that window into memory
+   * for future read calls. After all the data is read in, the ready callback will be invoked.
+   * The ready callback will take in a SessionReadyMessage* that will contain the offset, the amount of bytes
+   * , and the buffer in the form of a vector<char>.
+   */
+  void startReadSession(File file, size_t bytes, size_t offset, CkCallback ready);
```

Comment:
The publci API should be added to the documentation

Reply:
Yeah the docs will be added after this gets merged presumably.

Reply:
ok, I'll get on that.